esx_service/cli/vmdkops_admin.py

```diff
@@ -1301,16 +1305,38 @@ def config_rm(args):
     # This asks for double confirmation, and removes the local link or DB (if any)
     # NEVER deletes the shared database - instead prints help
 
-    if not args.local:
+    if not args.local and not args.unlink:
         return err_out("""
         Shared DB removal is not supported. For removing  local configuration, use --local flag.
-        For removing shared DB,  run 'vmdkops_admin config rm --local' on ESX hosts using this DB,
+        For removing shared DB,  run 'vmdkops_admin config rm --unlink' on ESX hosts using this DB,
         and manually remove the {} file from shared storage.
         """.format(auth_data.CONFIG_DB_NAME))
 
+    # Check the existing config mode
+    with auth_data.AuthorizationDataManager() as auth:
+        try:
+            auth.connect()
+            info = auth.get_info()
+            mode = auth.mode # for usage outside of the 'with'
+        except auth_data.DbAccessError as ex:
```

Comment:
Here we cannot let the code go through. When the exception happens, we cannot get the correct config mode from DB. And our following code depends on the config mode. We have similar logic in the "config_init()" too.